dist/index.js

```diff
@@ -6710,29 +6710,81 @@ const os = __nccwpck_require__(2037);
     // Install related tools.
     // Hopefully in the future we can install Dafny itself this way as well.
     // For now the zipped releases are simpler because they include Z3.
-    await installDotnetTool("dafny-reportgenerator", "1.*")
+    await installDotnetTool("dafny-reportgenerator", "1.*");
   } catch (error) {
     core.setFailed(error.message);
   }
 })();
 
 async function installDotnetTool(toolName, version) {
-  await exec.exec("dotnet", ["tool", "install", "-g", toolName, "--version", version])
+  await exec.exec("dotnet", [
+    "tool",
+    "install",
+    "-g",
+    toolName,
+    "--version",
+    version,
+  ]);
 }
 
 // Export functions for testing
 exports.dafnyURL = dafnyURL;
 exports.getDistribution = getDistribution;
 
-function dafnyURL(version, distribution) {
+async function dafnyURL(version, distribution) {
   const versionPath = version.startsWith("nightly") ? "nightly" : `v${version}`;
+  if (version == "nightly-latest") {
+    version = await latestNightlyVersion();
+  }
   const root = "https://github.com/dafny-lang/dafny/releases/download";
-  return `${root}/${versionPath}/dafny-${version == "2.3.0" ? "2.3.0.10506" : version}-x64-${distribution}.zip`;
+  return `${root}/${versionPath}/dafny-${
+    version == "2.3.0" ? "2.3.0.10506" : version
+  }-x64-${distribution}.zip`;
+}
+
+async function latestNightlyVersion() {
+  // Shamelessly copied and modified from dafny-lang/ide-vscode
+  // I'd prefer to use the GitHub API to list the assets under the "nightly" release,
+  // but @actions/github requires authentication.
```

Comment:
You are running in a GitHub action so you are authenticated.

Reply:
Oh I suppose I can use GITHUB_TOKEN, true. Not sure how to refactor the tests to still work though.